# Health Stats Part 4: Waist 2 Hip Ratios - Pandas Only

<!--- Write an explanation of the Waist To Hips Ratio statistic used by health professionals. Please include an explanation of what it is used for, exactly how it is calculated, and how to interpret the results. Note: Formmatting matters. Make this as professional as you can using Markdown.  --->

## W2H Ratio

The waist-hip ratio or waist-to-hip ratio (WHR) is the dimensionless ratio of the circumference of the waist to that of the hips. This is calculated as waist measurement divided by hip measurement (W ÷ H). For example, a person with a 30″ (76 cm) waist and 38″ (97 cm) hips has a waist-hip ratio of about 0.78.
## The Ratio is
$ ratio_{w2h} = \frac{w}{h} $

## People are separated into Either Apple or Pear Shapped
<table class="table table-responsive table-bordered table-striped table-hover">
            <thead>
                <tr class="info">
                    <th>Men</th>
                    <th>Women</th>
                    <th>Body Shape</th>
                    <th>Risk Level</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td>
                        Less than 0.90
                    </td>
                    <td>
                        Less than 0.85
                    </td>
                    <td>Pear</td>
                    <td>Low risk of metabolic complications</td>
                </tr>
                <tr>
                    <td>
                        Between 0.90 and 0.95
                    </td>
                    <td>
                        Between 0.85 and 0.90
                    </td>
                    <td>Apple</td>
                    <td>Moderate risk of metabolic complications like diabetes, heart disease, stroke, etc..</td>
                </tr>
                <tr>
                    <td>
                        Above 0.95
                    </td>
                    <td>
                        Above 0.90
                    </td>
                    <td>Apple</td>
                    <td>High risk of metabolic complications like diabetes, heart disease, stroke, etc..</td>
                </tr>
            </tbody>
        </table>
        
*link: https://www.mytecbits.com/tools/medical/whrcalculator*

## Source Data 

<!--- Replace the text below with a Markdown bullet list that defines the columns of the CSV file. Be sure to indicate the data type for each column. --->
EDIT THIS MARKDOWN CELL

- Waist - int
- Hip - int
- Gender - str
- Shape - int

## Data Import

In [104]:
# Goal: Extract the data from the file

# use proper pandas function to read data from a CSV file to a DataFrame



## Calculations

In [11]:
# Goal: For each row of data calculate and store the w2h_ratio and shape.
import pandas as pd
import numpy as np
import io

df = pd.read_csv('w2h_data.csv')
df
# Step 1: Make sure the columns are in correct data types - refer to previous parts for the correct data types




ID  Waist  Hip Gender
0    1     30   32      M
1    2     32   37      M
2    3     30   36      M
3    4     33   39      M
4    5     29   33      M
5    6     32   38      M
6    7     33   42      M
7    8     30   40      M
8    9     30   37      M
9   10     32   39      M
10  11     24   35      F
11  12     25   37      F
12  13     24   37      F
13  14     22   34      F
14  15     26   38      F
15  16     26   37      F
16  17     25   38      F
17  18     26   37      F
18  19     28   40      F
19  20     23   35      F

In [31]:
# Step 2: calculate the 'W2H_Ratio' column using the 'Waist' and 'Hip' columns
# Step 3: Create the `Shape` column based on the values of the `W2H_Ratio` column - refer to previous parts for the logic

df['W2H_Ratio'] = df['Waist']/ df['Hip']
df['Shape'] = ['Apple' if d > 1 else "Pear" for d in df["W2H_Ratio"]]
df

ID  Waist  Hip Gender  W2H_Ratio Shape
0    1     30   32      M   0.937500  Pear
1    2     32   37      M   0.864865  Pear
2    3     30   36      M   0.833333  Pear
3    4     33   39      M   0.846154  Pear
4    5     29   33      M   0.878788  Pear
5    6     32   38      M   0.842105  Pear
6    7     33   42      M   0.785714  Pear
7    8     30   40      M   0.750000  Pear
8    9     30   37      M   0.810811  Pear
9   10     32   39      M   0.820513  Pear
10  11     24   35      F   0.685714  Pear
11  12     25   37      F   0.675676  Pear
12  13     24   37      F   0.648649  Pear
13  14     22   34      F   0.647059  Pear
14  15     26   38      F   0.684211  Pear
15  16     26   37      F   0.702703  Pear
16  17     25   38      F   0.657895  Pear
17  18     26   37      F   0.702703  Pear
18  19     28   40      F   0.700000  Pear
19  20     23   35      F   0.657143  Pear

In [34]:
str_io = io.StringIO()

df.to_html(buf=str_io, classes='table table-striped')

html_str = str_io.getvalue()

print(html_str)


<table border="1" class="dataframe table table-striped">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ID</th>
      <th>Waist</th>
      <th>Hip</th>
      <th>Gender</th>
      <th>W2H_Ratio</th>
      <th>Shape</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>30</td>
      <td>32</td>
      <td>M</td>
      <td>0.937500</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>32</td>
      <td>37</td>
      <td>M</td>
      <td>0.864865</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>30</td>
      <td>36</td>
      <td>M</td>
      <td>0.833333</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>33</td>
      <td>39</td>
      <td>M</td>
      <td>0.846154</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>29</td>
      <td>33</td>
      <td>M</td>
      <td>0.878788</td>
    

<table border="1" class="dataframe table table-striped">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ID</th>
      <th>Waist</th>
      <th>Hip</th>
      <th>Gender</th>
      <th>W2H_Ratio</th>
      <th>Shape</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>30</td>
      <td>32</td>
      <td>M</td>
      <td>0.937500</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>32</td>
      <td>37</td>
      <td>M</td>
      <td>0.864865</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>30</td>
      <td>36</td>
      <td>M</td>
      <td>0.833333</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>33</td>
      <td>39</td>
      <td>M</td>
      <td>0.846154</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>29</td>
      <td>33</td>
      <td>M</td>
      <td>0.878788</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>5</th>
      <td>6</td>
      <td>32</td>
      <td>38</td>
      <td>M</td>
      <td>0.842105</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>6</th>
      <td>7</td>
      <td>33</td>
      <td>42</td>
      <td>M</td>
      <td>0.785714</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8</td>
      <td>30</td>
      <td>40</td>
      <td>M</td>
      <td>0.750000</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>8</th>
      <td>9</td>
      <td>30</td>
      <td>37</td>
      <td>M</td>
      <td>0.810811</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>9</th>
      <td>10</td>
      <td>32</td>
      <td>39</td>
      <td>M</td>
      <td>0.820513</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>10</th>
      <td>11</td>
      <td>24</td>
      <td>35</td>
      <td>F</td>
      <td>0.685714</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>11</th>
      <td>12</td>
      <td>25</td>
      <td>37</td>
      <td>F</td>
      <td>0.675676</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>12</th>
      <td>13</td>
      <td>24</td>
      <td>37</td>
      <td>F</td>
      <td>0.648649</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>13</th>
      <td>14</td>
      <td>22</td>
      <td>34</td>
      <td>F</td>
      <td>0.647059</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>14</th>
      <td>15</td>
      <td>26</td>
      <td>38</td>
      <td>F</td>
      <td>0.684211</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>15</th>
      <td>16</td>
      <td>26</td>
      <td>37</td>
      <td>F</td>
      <td>0.702703</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>16</th>
      <td>17</td>
      <td>25</td>
      <td>38</td>
      <td>F</td>
      <td>0.657895</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>17</th>
      <td>18</td>
      <td>26</td>
      <td>37</td>
      <td>F</td>
      <td>0.702703</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>18</th>
      <td>19</td>
      <td>28</td>
      <td>40</td>
      <td>F</td>
      <td>0.700000</td>
      <td>Pear</td>
    </tr>
    <tr>
      <th>19</th>
      <td>20</td>
      <td>23</td>
      <td>35</td>
      <td>F</td>
      <td>0.657143</td>
      <td>Pear</td>
    </tr>
  </tbody>
</table>

## Output
<table border="1" class="dataframe table table-striped">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>ID</th>
      <th>Waist</th>
      <th>Hip</th>
      <th>Gender</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>30</td>
      <td>32</td>
      <td>M</td>
    </tr>
    <tr>
      <th>1</th>
      <td>2</td>
      <td>32</td>
      <td>37</td>
      <td>M</td>
    </tr>
    <tr>
      <th>2</th>
      <td>3</td>
      <td>30</td>
      <td>36</td>
      <td>M</td>
    </tr>
    <tr>
      <th>3</th>
      <td>4</td>
      <td>33</td>
      <td>39</td>
      <td>M</td>
    </tr>
    <tr>
      <th>4</th>
      <td>5</td>
      <td>29</td>
      <td>33</td>
      <td>M</td>
    </tr>
    <tr>
      <th>5</th>
      <td>6</td>
      <td>32</td>
      <td>38</td>
      <td>M</td>
    </tr>
    <tr>
      <th>6</th>
      <td>7</td>
      <td>33</td>
      <td>42</td>
      <td>M</td>
    </tr>
    <tr>
      <th>7</th>
      <td>8</td>
      <td>30</td>
      <td>40</td>
      <td>M</td>
    </tr>
    <tr>
      <th>8</th>
      <td>9</td>
      <td>30</td>
      <td>37</td>
      <td>M</td>
    </tr>
    <tr>
      <th>9</th>
      <td>10</td>
      <td>32</td>
      <td>39</td>
      <td>M</td>
    </tr>
    <tr>
      <th>10</th>
      <td>11</td>
      <td>24</td>
      <td>35</td>
      <td>F</td>
    </tr>
    <tr>
      <th>11</th>
      <td>12</td>
      <td>25</td>
      <td>37</td>
      <td>F</td>
    </tr>
    <tr>
      <th>12</th>
      <td>13</td>
      <td>24</td>
      <td>37</td>
      <td>F</td>
    </tr>
    <tr>
      <th>13</th>
      <td>14</td>
      <td>22</td>
      <td>34</td>
      <td>F</td>
    </tr>
    <tr>
      <th>14</th>
      <td>15</td>
      <td>26</td>
      <td>38</td>
      <td>F</td>
    </tr>
    <tr>
      <th>15</th>
      <td>16</td>
      <td>26</td>
      <td>37</td>
      <td>F</td>
    </tr>
    <tr>
      <th>16</th>
      <td>17</td>
      <td>25</td>
      <td>38</td>
      <td>F</td>
    </tr>
    <tr>
      <th>17</th>
      <td>18</td>
      <td>26</td>
      <td>37</td>
      <td>F</td>
    </tr>
    <tr>
      <th>18</th>
      <td>19</td>
      <td>28</td>
      <td>40</td>
      <td>F</td>
    </tr>
    <tr>
      <th>19</th>
      <td>20</td>
      <td>23</td>
      <td>35</td>
      <td>F</td>
    </tr>
  </tbody>
</table>

In [28]:
# Goal: pretty print the rows as an HTML table

# Display the complete DF
df
# Save the DF to a file './complete_w2h.csv'





ID  Waist  Hip Gender  W2H_Ratio
0    1     30   32      M   0.937500
1    2     32   37      M   0.864865
2    3     30   36      M   0.833333
3    4     33   39      M   0.846154
4    5     29   33      M   0.878788
5    6     32   38      M   0.842105
6    7     33   42      M   0.785714
7    8     30   40      M   0.750000
8    9     30   37      M   0.810811
9   10     32   39      M   0.820513
10  11     24   35      F   0.685714
11  12     25   37      F   0.675676
12  13     24   37      F   0.648649
13  14     22   34      F   0.647059
14  15     26   38      F   0.684211
15  16     26   37      F   0.702703
16  17     25   38      F   0.657895
17  18     26   37      F   0.702703
18  19     28   40      F   0.700000
19  20     23   35      F   0.657143

In [30]:
export_csv = df.to_csv (r'./complete_w2h.csv', index = None, header=True) 

## Lessons Learned

Have you noticed in the previous two parts, how much code we have written? Have you also note how little code we have written in this part, with the help of Pandas? 

This is the reason why we want to use Pandas to handle the data we use for analytics.